packages/material-ui/src/FormControl/FormControl.js

```diff
@@ -48,6 +48,13 @@ export const styles = {
  * ⚠️ Only one input can be used within a FormControl.
  */
 class FormControl extends React.Component {
+  static getDerivedStateFromProps(props, state) {
```

Comment:
I think that it's the InputBase that should take care of it by calling `handleBlur`.

Reply:
```diff
- <FormControl disabled={false}><InputBase /></FormControl>
+ <FormControl disabled={true}><InputBase /></FormControl>
```

InputBase already calls handleBlur separately if it's own focused state changes.

Reply:
I don't follow, the disable change will propagate down to the input.

Reply:
Maybe we need an integration test? :)

Reply:
It's context not props propagation. If you think that some behavior was already covered feel free to add a test. Especially focused was not considered correctly.

But why should InputBase call onBlur in it's parent when the parent got changed and can take care of itself? No need to delegate that logic into the child.

Reply:
The form control should propagate the disabled state to the input with the context. The input can then collect it. This way, we can avoid logic duplication. I think that we should remove the old context usage at the same time this logic is very dependant on the context implementation. I think that it will be simpler and less error prone to do both at the same time.

Yeah, it would be great to have an integration tests. It's tricky. Well, I'm also fine with logic duplication if it's simpler.

Reply:
It's only logic duplication if you have a mental model with the old context and lifecycle methods. I don't think the discussion is helpful here. If you know how to access context in gDSFP or cDU I'd be happy to change it but I don't know how.

>  this logic is very dependant on the context implementation. 
I can only work with what's been given to me. InputBase was always tightly coupled with the context.

Reply:
> gDSFP or cDU I'd be happy to change it but I don't know how.

We can't, it's why I'm raising the next context api, where we would injecting the context as properties.